content/en/platform/corda/4.10/enterprise/node/operating/monitoring-and-logging/opentelemetry.md

```diff
@@ -0,0 +1,103 @@
+
+## Client API
+
+OpenTelemetry span generation is also incorporated into the RPC client API. If you want your Corda client code to generate spans, you need to include the OpenTelemetry SDK as a dependency. The methods described above only apply to the Corda node. 
+
+<!-- (TODO: Give example of dependencies needed). -->
+
+## Creating your own Spans
+
+The OpenTelemetry API may be used in your flows and in your client code to create spans and baggage. To get an instance of the OpenTelemetry API in a flow, make the following call:
+`val openTelemetry: OpenTelemetry? = serviceHub.telemetryService.getTelemetryHandle(OpenTelemetry::class.java)`.
+
+From the client API, where RPC is a CordaRPCConnection, you would use the following:
+`val openTelemetry: OpenTelemetry? = rpc.getTelemetryHandle(OpenTelemetry::class.java)`.
+
+When creating your own spans, you can also create your own baggage. If you create your own baggage, it will also be sent to other nodes, and you can specify if you want this baggage to be copied to span tags. If you do, all of the spans involved in the transaction for that node will also get a copy of the baggage. This can be enabled with the following parameter:
+`telementry.copyBaggageToTags = true`.
+
+The default value of this setting is `false`.
+
+{{< note >}}
+Within a checkpoint, only the span ID has a checkpoint, meaning spans do not survive a node restart. If the node restarts, the parent span information will be lost, and new spans will be generated for the flows. The root span after the node restart won't know who the parent span was before the node restart.
+{{< /note >}}
+
+## Start and End Spans
+
+The current implementation of OpenTelemetry will send spans to the backend when the flow or operation is completed. This is handled via the OpenTelemetry SDK. If the flow somehow gets stuck or does not complete, the span representing that flow will never reach the backend. It can be difficult to figure out what went wrong by just looking at the spans, as you will only see the complete spans.
+
+{{< note >}}
+If a child span doesn't complete, parent spans also do not complete.
+{{< /note >}}
+
+As an alternative, the Corda OpenTelemetry component sends a start or end span to the backend when a flow or operation starts or stops, in addition to the normal spans sent for the operation. This is effectively a start flow span event and an end flow span event. With this view of the spans, it becomes easier to determine where the flow got stuck, as it will be the lowest child without an end span event. 
+
+{{< note >}}
+These start and end span events are only generated for spans that Corda generated. If you create a span in your own flow code, you won’t see equivalent start and end span events for your flows, as Corda knows nothing of them before the node restart.
```

Comment:
Can you delete "before the node restart" at the end of this sentence.

Reply:
Done